% Generated by roxygen2 (4.0.2): do not edit by hand
\name{simulate_experiment_countmat}
\alias{simulate_experiment_countmat}
\title{Simulate RNA-seq experiment}
\usage{
simulate_experiment_countmat(fasta = NULL, gtf = NULL, seqpath = NULL,
  readmat, outdir = ".", fraglen = 250, fragsd = 25, readlen = 100,
  error_rate = 0.005, paired = TRUE, seed = NULL, ...)
}
\arguments{
\item{fasta}{path to FASTA file containing transcripts from which to simulate
reads. See details.}

\item{gtf}{path to GTF file containing transcript structures from which reads
should be simulated. See details.}

\item{seqpath}{path to folder containing one FASTA file (\code{.fa}
extension) for each chromosome in \code{gtf}. See details.}

\item{readmat}{matrix with rows representing transcripts and columns
representing samples. Entry i,j specifies how many reads to simulate from
transcript i for sample j.}

\item{outdir}{character, path to folder where simulated reads should be
written, without a slash at the end of the folder name. By default, reads
written to the working directory.}

\item{fraglen}{Mean RNA fragment length. Sequences will be read off the
end(s) of these fragments.}

\item{fragsd}{Standard deviation of fragment lengths.}

\item{readlen}{Read length}

\item{error_rate}{Sequencing error rate. Must be between 0 and 1. A uniform
error model is assumed.}

\item{paired}{If \code{TRUE}, paired-end reads are simulated; else single-end
reads are simulated.}

\item{seed}{Optional seed to set before simulating reads, for
reproducibility.}

\item{...}{Further arguments to pass to \code{seq_gtf}, if \code{gtf} is not
\code{NULL}.}
}
\value{
No return, but simulated reads are written to \code{outdir}.
}
\description{
create FASTA files containing RNA-seq reads simulated from provided
  transcripts, with optional differential expression between two groups
  (designated via read count matrix)
}
\details{
Reads can either be simulated from a FASTA file of transcripts
  (provided with the \code{fasta} argument) or from a GTF file plus DNA
  sequences (provided with the \code{gtf} and \code{seqpath} arguments).
  Simulating from a GTF file and DNA sequences may be a bit slower: it took
  about 6 minutes to parse the GTF/sequence files for chromosomes 1-22,
  X, and Y in hg19.
}
\examples{
\donttest{
  fastapath = system.file("extdata", "chr22.fa", package="polyester")
  numtx = count_transcripts(fastapath)
  readmat = matrix(20, ncol=10, nrow=numtx)
  readmat[1:30, 1:5] = 40

  simulate_experiment_countmat(fasta=fastapath,
    readmat=readmat, outdir='simulated_reads_2', seed=5)
}
}

